src/dc_receive_imf.rs

```diff
@@ -2720,4 +2561,75 @@ mod tests {
             "$BLOBDIR/test pdf äöüß.pdf"
         );
     }
+
+    /// Test that classical MUA messages are assigned to group chats based on the `In-Reply-To`
+    /// header.
+    #[async_std::test]
+    async fn test_in_reply_to() {
+        let t = TestContext::new().await;
+        t.configure_addr("bob@example.com").await;
+
+        // Receive message from Alice about group "foo".
+        dc_receive_imf(
+            &t,
+            b"Received: (Postfix, from userid 1000); Mon, 4 Dec 2006 14:51:39 +0100 (CET)\n\
+                 From: alice@example.org\n\
+                 To: bob@example.com, charlie@example.net\n\
+                 Subject: foo\n\
+                 Message-ID: <message@example.org>\n\
+                 Chat-Version: 1.0\n\
+                 Chat-Group-ID: foo\n\
+                 Chat-Group-Name: foo\n\
+                 Date: Sun, 22 Mar 2020 22:37:57 +0000\n\
+                 \n\
+                 hello foo\n",
```

Comment:
lines are really meant to end with `\r\n` instead of just `\n`, but I guess our parser is lenient...  still

Reply:
I copied this from other tests. If we want to change this, then it makes sense to change in all places.